force-app/main/default/classes/RD2_OpportunityEvaluation_BATCH.cls

```diff
@@ -225,6 +229,24 @@ public without sharing class RD2_OpportunityEvaluation_BATCH
         }
     }
 
+    private Integer getBatchSize() {
+        Decimal configValue = UTIL_CustomSettingsFacade.getRecurringDonationsSettings().Recurring_Donation_Batch_Size__c;
+
+        if (configValue == null) {
+            configValue = 50;
+        }
+
+        if (RD2_ElevateIntegrationService.isIntegrationEnabled() &&
+            (rd2Settings.isStatusAutomationForClosedEnabled() || rd2Settings.isStatusAutomationForLapsedEnabled())
+        ) {
+            Decimal maxSize = RD2_UpdateCommitmentBulkService.MAXIMUM_API_CALL_PER_TRANSACTION
```

Comment:
@lparrott I made some small adjustments to increase code readability. BatchSize is only called from `Database.executeBatch({{your-batch}}, batchSize)` to determine the record count of individual transactions in a batch. In this case, I am limiting the max record count to 250 records (5 * 50) when the Bulk API can be performed. The reasons for 250 records are the following:
-  The maximum record count per API callout is 50 records
-  The recommended timeout per API callout is set to 20 seconds and the government limit on API timeout per transaction is 120 seconds. 5 API callout will lend at 100 seconds on API timeout. I give a 20 seconds buffer room just in case.

Therefore, the bulk commitment API MAXIMUM_API_CALL_PER_TRANSACTION * REQUEST_SIZE get us 250 records.